src/Libraries/CoreNodes/List.cs

```diff
@@ -782,6 +782,38 @@ public static IList Combinations(IList list, int length, bool replace = false)
                     .ToList();
         }
 
+        /// <summary>
+        ///     Return the first index of item in the given list. 
+        ///     Return -1 if not found.
+        /// </summary>
+        /// <param name="list">List to search in</param>
+        /// <param name="item">Item to look for</param>
+        /// <returns></returns>
+        public static int FirstIndexOf(IList list, object item)
+        {
+            if (list == null)
+                return -1;
+
+            int index = list.IndexOf(item);
+            return index;
+        }
+
+        /// <summary>
+        ///     Return all indice of item in the given list.
+        ///     Return empty list if not found.
+        /// </summary>
+        /// <param name="list"></param>
+        /// <param name="item"></param>
+        /// <returns></returns>
+        public static IList AllIndicesOf(IList list, object item)
+        {
+            if (list == null)
+                return new List<int> { }; 
+
+            var indices = Enumerable.Range(0, list.Count).Where(i => list[i].Equals(item)).ToList();
+            return indices;
+        }
+
```

Comment:
@kronz , do we need replication on the second parameter of `List.FirstIndexOf()` and `List.AllIndicesOf()`? For example, should `List.FirstIndexOf({1, 2, 1, 2}, {1, 2})` return `0`, or return `{0, 1}`?